Refactored RecordPage.xaml.cs into separate controller classes

treytipton opened the pull request:
- Split main CS file into multiple files for better readability and maintainability
- Improves debugging and simplifies future changes

Forgive me for the huge change I was trying to make everything easier to read and modify later on down the road.

treytipton committed:
Refactored RecordPage.xaml.cs into separate controller classes

- Split main CS file into multiple files for better readability and maintainability
- Improves debugging and simplifies future changes

rmilewski4 commented:
This app version hangs on the loading screen when I try to open it. @leroyperkins can you try to run it on your machine?

treytipton committed:
Added video recording functionality

Recording a launch now records an mp4 to export with the .json in a timestamped folder

treytipton committed:
Added targeted frame rate capture

Adds delays that accommodate frame processing time
to accurately match target frame rate

treytipton committed:
reduced heavy lifting of recordpage constructor

Takes a lot of the heavy lifting off of the UI thread and moves it onto
background threads instead to allow the UI to actually be able to load.
Also fixed a bunch of build warnings in this patch as well.

treytipton committed:
Added new library to simplify code

Added OpenCvSharp.WpfExtensions to simplify WebcamManager.cs.

Removes unsafe code (DeleteObject)

treytipton committed:
Update packages.

treytipton committed:
Fixed bug that made "Loading..." overlay over webcam footage.

treytipton commented:
Fixed the unresponsive loading screen, turns out we had a bunch of stuff being loaded on the UI thread which didn't allow it to be updated. Moved all of the Webcam, Labjack, and other managers we were using onto background threads.

leroyperkins reviewed (approved):
LGTM!
